packages/ui/src/theme/__tests__/createTheme.test.ts

```diff
@@ -33,36 +34,52 @@ describe('@aws-amplify/ui', () => {
       });
     });
 
-    describe('with a theme and without a base theme', () => {
+    describe('with a custom theme using token objects and without a base theme', () => {
+      const shadows = {
+        small: { value: '12px' },
+        large: {
+          value: {
+            offsetX: 'offsetX',
+            offsetY: 'offsetY',
+            blurRadius: 'blurRadius',
+            spreadRadius: 'spreadRadius',
+            color: 'color',
+          },
+        },
+      };
       const theme = createTheme({
         name: 'test-theme',
         tokens: {
           colors: {
-            background: {
-              primary: { value: '#bada55' },
-            },
-            font: {
-              primary: { value: '{colors.white.value}' },
-            },
+            background: { primary: { value: '#bada55' } },
+            font: { primary: { value: '{colors.white.value}' } },
           },
+          components: { alert: { alignItems: { value: 'right' } } },
+          shadows,
         },
       });
 
       it('should override the base theme', () => {
         const { tokens } = theme;
         expect(tokens.colors.background.primary.value).toEqual('#bada55');
+
+        expect(tokens.shadows.large.value).toEqual(
+          'offsetX offsetY blurRadius spreadRadius color'
+        );
+        expect(tokens.shadows.small.value).toEqual(shadows.small.value);
+
+        expect(
+          // TODO: link gthub issue
+          (tokens.components.alert.alignItems as WebDesignToken).value
```

Comment:
For properties of the `components` key  of `BaseTheme` to accept both syntaxes, `DesignToken` was extended [here](https://github.com/aws-amplify/amplify-ui/pull/2902/files#diff-061a7577867c5583dc1ceeafba7707e260d62c775c6f950bc893f9f463f3182cR4). 

We need something along the lines of a `WebComponents` type (similar to our `WebTokens` and other `Web` prefixed types) that will be used as the output for `components` from `createTheme`. Right now there is only the `components` key on `Tokens` being shared between both `Tokens` and `WebTokens`, and it ultimately uses `DesignToken` for the style properties to allow both custom style syntaxes, but the output of `createTheme` creates the `components` style properties only as `WebDesignToken` but without the correct typing. 

Would prefer to address this issue in a follow up as creating a `WebComponents` type will touch every component type file at the very least, this issue is internal, and is not visible any where in the codebase except for this file.

Reply:
Actually think the issue described above needs to be addressed as a part of this PR since there is a potential edge case where a developer could be referencing a `component` style property in their own code resulting in a type error, example:
```
<View style={{ alignItems: theme.tokens.components.alert.alignItems.value }} />
```

Reply:
[Pushed a commit](https://github.com/aws-amplify/amplify-ui/pull/2902/commits/a0c328f13e8f5de14ee4ade37dd887a08fcd4da6) with `Alert`, `Authenticator`, `Badge` and `Card`  refactored